client/snap_op.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"fmt"
+	"strings"
+)
+
+// AddSnap adds the snap with the given name, returning the UUID of the
+// background operation upon success.
+func (client *Client) AddSnap(name string) (string, error) {
+	path := fmt.Sprintf("/2.0/snaps/%s", name)
+	body := strings.NewReader(`{"action":"install"}`)
+
+	return client.doAsync("POST", path, nil, body)
+}
+
+// RemoveSnap removes the snap with the given name, returning the UUID of the
+// background operation upon success.
+func (client *Client) RemoveSnap(name string) (string, error) {
+	path := fmt.Sprintf("/2.0/snaps/%s", name)
+	body := strings.NewReader(`{"action":"remove"}`)
+
+	return client.doAsync("POST", path, nil, body)
+}
+
+// RefreshSnap refreshs the snap with the given name, returning the UUID of the
```

Comment:
This change makes me realize that several of the function comments here just repeat the method name (Foo fooes), which is not very helpful. It'd be nice to provide some further guidance at some point about what these methods actually do in more detail.